sources/src/main/java/com/google/solutions/jitaccess/core/services/RoleDiscoveryService.java

```diff
@@ -291,9 +291,7 @@ public Result<ProjectRole> listEligibleProjectRoles(
     consolidatedRoles.addAll(activatedRoles);
 
     return new Result<>(
-      consolidatedRoles.stream()
-        .sorted(Comparator.comparing(r -> r.roleBinding.fullResourceName))
-        .collect(Collectors.toList()),
+      consolidatedRoles.stream().sorted().collect(Collectors.toList()),
```

Comment:
The `sorted()` call destroys the order again, you probably have to remove this.

This is how it looks now -- it's sorted across all 3 buckets as opposed to within each bucket:
![image](https://github.com/GoogleCloudPlatform/jit-access/assets/5724941/5a365f25-9a0d-4ee9-8ee8-02efdb667e2c)

Reply:
Fixed. Yep, now that additional sorting can be removed as I understand.